src/hotspot/share/services/heapDumper.cpp

```diff
@@ -1164,7 +1146,7 @@ void DumperSupport::dump_instance(AbstractDumpWriter* writer, oop o, DumperClass
 
   DumperClassCacheTableEntry* cache_entry = class_cache->lookup_or_create(ik);
 
-  u4 is = instance_size(ik, cache_entry);
+  u4 is = cache_entry->instance_size();
```

Comment:
The old version had a check  `(cache_entry != nullptr)` at line 1046.
Is it not applicable anymore? Can the `cache_entry` be set to `nullptr` at this point?